map/src/util/Utils.js

```diff
@@ -111,24 +115,38 @@ async function uploadFile(gpxFiles, setGpxFiles, ctx, gpxLayer, file) {
 
 function getFileName(currentFile) {
     let file = Object.assign('', currentFile);
-    file.name = file.name.replace(/.gpx/, '');
-    if (file.name.includes('/')) {
-        return file.name.split('/')[1]
-    } else if (file.local && file.name.includes(':')) {
-        return file.name.split(':')[1]
+    return prepareName(file.name, file.local);
+}
+
+function prepareName(name, local) {
```

Comment:
Bad function for global utility probably should not be here. 
Global utility should have only common algorithms no business logic like name formatting by unknown rules.
Good Function:
- stripExtension(fileName);

Bad Function
- formatGpxName(fileName);